### ISSUE TITLE
New restart API fails for CV array container

### ISSUE THREAD

**Issue by MTCam** [illinois-ceesd/mirgecom#398]
When trying to read a restart file with restart.py `read_restart_data`, the error pasted in below occurs.  A reproducing example is demonstrated in #396.

```
  File "autoignition-mpi.py", line 116, in main
    restart_data = read_restart_data(actx, rst_filename)
  File "/Users/mtcampbe/CEESD/devel/gamma/emirge/mirgecom/mirgecom/restart.py", line 41, in read_restart_data
    return pickle.load(f)
  File "/Users/mtcampbe/CEESD/devel/gamma/emirge/meshmode/meshmode/dof_array.py", line 283, in __setstate__
    self.array_context = actx
AttributeError: can't set attribute
```

**Comment by inducer** [comment]
Should be fixed by https://github.com/inducer/meshmode/pull/234 (once it goes in).

**Comment by MTCam** [comment]
> Should be fixed by [inducer/meshmode#234](https://github.com/inducer/meshmode/pull/234) (once it goes in).

Thanks a ton for fixing this up so quickly, @inducer.  I tweaked up #396 to add a simple test that demonstrates that this issue is fixed, and will close as soon as it passes.
